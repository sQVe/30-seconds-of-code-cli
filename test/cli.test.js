const path = require('path');
const { spawnSync } = require('child_process');

const { version } = require('../package.json');

const callCli = (...args) =>
  spawnSync(path.resolve(__dirname, '../dist/cli.js'), args)
    .stdout.toString()
    .trim();
const commands = ['search', 'view', 'tag'];

describe('Cli', () => {
  describe('version option', () => {
    it('should print the package version', async () => {
      expect(callCli('-V')).toEqual(version);
      expect(callCli('--version')).toEqual(version);
    });
  });

  describe('help option', () => {
    it('should print help page', async () => {
      expect(callCli('-h')).toMatchSnapshot();
      expect(callCli('--help')).toMatchSnapshot();
    });

    commands.forEach(command => {
      it(`should print help page for ${command} command`, () => {
        expect(callCli(command, '-h')).toMatchSnapshot();
        expect(callCli(command, '--help')).toMatchSnapshot();
      });
    });
  });

  commands.forEach(command => {
    describe(`${command} command`, () => {
      it(`should output test strings`, async () => {
        expect(callCli(command, 'foo')).toEqual(
          JSON.stringify([command, 'foo', 'iced', false])
        );
        expect(callCli(command, 'foo', '--layout', 'ic')).toEqual(
          JSON.stringify([command, 'foo', 'ic', false])
        );
        expect(callCli(command, 'foo', '--json')).toEqual(
          JSON.stringify([command, 'foo', 'iced', true])
        );
      });
    });
  });
});
