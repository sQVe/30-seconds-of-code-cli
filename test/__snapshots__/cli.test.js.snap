// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cli help option should print help page 1`] = `
"Usage: cli [options] [command]

Options:
  -V, --version             output the version number
  -h, --help                output usage information

Commands:
  search [options] [query]  Fuzzy search snippets
  tag [options] [id]        View snippets related to tag
  view [options] [id]       View snippet

Examples:
  view all
  search flatten --output json all
  tag --print ic array"
`;

exports[`Cli help option should print help page 2`] = `
"Usage: cli [options] [command]

Options:
  -V, --version             output the version number
  -h, --help                output usage information

Commands:
  search [options] [query]  Fuzzy search snippets
  tag [options] [id]        View snippets related to tag
  view [options] [id]       View snippet

Examples:
  view all
  search flatten --output json all
  tag --print ic array"
`;

exports[`Cli help option should print help page for search command 1`] = `
"Usage: search [options] [query]

Fuzzy search snippets

Options:
  -l, --layout <layout>  print in specified layout (default: \\"iced\\")
  -c, --cp               copy code blocks
  -j, --json             output in json format
  -h, --help             output usage information"
`;

exports[`Cli help option should print help page for search command 2`] = `
"Usage: search [options] [query]

Fuzzy search snippets

Options:
  -l, --layout <layout>  print in specified layout (default: \\"iced\\")
  -c, --cp               copy code blocks
  -j, --json             output in json format
  -h, --help             output usage information"
`;

exports[`Cli help option should print help page for tag command 1`] = `
"Usage: tag [options] [id]

View snippets related to tag

Options:
  -l, --layout <layout>  print in specified layout (default: \\"iced\\")
  -c, --cp               copy code blocks
  -j, --json             output in json format
  -h, --help             output usage information"
`;

exports[`Cli help option should print help page for tag command 2`] = `
"Usage: tag [options] [id]

View snippets related to tag

Options:
  -l, --layout <layout>  print in specified layout (default: \\"iced\\")
  -c, --cp               copy code blocks
  -j, --json             output in json format
  -h, --help             output usage information"
`;

exports[`Cli help option should print help page for view command 1`] = `
"Usage: view [options] [id]

View snippet

Options:
  -l, --layout <layout>  print in specified layout (default: \\"iced\\")
  -c, --cp               copy code blocks
  -j, --json             output in json format
  -h, --help             output usage information"
`;

exports[`Cli help option should print help page for view command 2`] = `
"Usage: view [options] [id]

View snippet

Options:
  -l, --layout <layout>  print in specified layout (default: \\"iced\\")
  -c, --cp               copy code blocks
  -j, --json             output in json format
  -h, --help             output usage information"
`;
